covid19-etl/etl/jhu_to_s3.py

```diff
@@ -801,6 +838,61 @@ def nested_dict_to_time_series_by_level(self):
                 os.remove(abs_path)
         print("  Done in {} secs".format(int(time.time() - start)))
 
+    def nested_dict_to_data_by_time(self):
+        """
+        See `nested_dict_to_geojson` docstring for details on the aggregation.
+        """
+        print("Generating {}...".format(IL_JSON_BY_TIME_FILENAME))
+        countyList = {}
+        for country_data in self.nested_dict.values():
+
+            # for US only
+            if country_data["country_region"] != "US":
+                continue
+
+            for province_data in country_data.get("provinces", {}).values():
+
+                # for IL only
+                if province_data["province_state"] != "Illinois":
+                    continue
+
+                for county_data in province_data.get("counties", {}).values():
+                    county_fips = county_data["fips"]
+
+                    countyList[county_fips] = {
+                        "county": county_data["county"],
+                        "by_date": {},
+                    }
+                    lastValue = {}
+                    # add county-level time_series data for IL counties
+                    for date, ts in county_data["time_series"].items():
+                        confirmed = ts.get("confirmed", 0)
+                        deaths = ts.get("deaths", 0)
+                        tempObj = {}
+
+                        if confirmed >= MINIMUM_COUNT:
+                            tempObj["C"] = replace_small_counts_simple(confirmed)
+
+                        if deaths >= MINIMUM_COUNT:
+                            tempObj["D"] = replace_small_counts_simple(deaths)
+
+                        if tempObj and lastValue != tempObj:
+                            # store last value to eliminate duplicate data and reduce file size
+                            lastValue = tempObj
+                            # add this US county. it shouldn't already be there
+                            countyList[county_fips]["by_date"][date] = tempObj
+
+        with open(
+            os.path.join(CURRENT_DIR, MAP_DATA_FOLDER, IL_JSON_BY_TIME_FILENAME), "w"
+        ) as f:
+            # create smaller file size by eliminating white space
+            f.write(
+                json.dumps(
+                    {"il_county_list": countyList, "last_updated": self.latest_date},
+                    separators=(",", ":"),
```

Comment:
maybe we should add this to the other json.dumps as well? 😄

Reply:
ill make a new PR for that once this is done